File: Dataset/Leetcode/valid/88/111.cpp
 class Solution {
public:
    void XXX(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if(n==0)
            return ;
        if(m==0){
            nums1 = nums2;
            return ;
        }
        int p = m-1;
        int q = n-1;
        int i = m+n-1;
        for(;i>=0;--i){
            if(nums1[p]>=nums2[q]){
                nums1[i] = nums1[p];
                p--;
            }
            else{
                nums1[i] = nums2[q];
                q--;
            }
            if(p<0){
                while(--i>=0)
                    nums1[i] = nums2[q--];
                return;
            }
            if(q<0)
                return;
        }
    }
};

File: Dataset/Leetcode/valid/14/111.java
 class Solution {
    public String XXX(String[] strs) {
        // 100%,89%
        // if(strs.length==0||strs==null)return "";
        // String str=strs[0];
        // int left=0,right=str.length();
        // for(int i=1;i<strs.length;++i){
        //     if(!strs[i].startsWith(str)){
        //         if(right-- == 0)return "";
        //         i--;
        //         str=strs[0].substring(0,right);
        //     }
        // }
        // return str;

        // 88.35%,98.90%
        if(strs.length==0||strs==null)return "";
        char[] chr=strs[0].toCharArray();
        int right=strs[0].length();
        
        for(int i=1;i<strs.length;++i){
            right=check(chr,strs[i],right);
            if(right==0)return "";
        }
        return new String(chr,0,right);
    }
    
    public int check(char[] ch,String str,int right){
        int i=0;
        for(i=0;i<right&&i<str.length();++i){
            if(ch[i]!=str.charAt(i))return i;
        }
        return i;
    }
}```

